ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,225 @@
+        self.resource_name = "prometheus-k8s" or resource_name
+        self.client = client or get_client()
+        self.api_v1 = "/api/v1"
+        self.verify_ssl = verify_ssl
+        self.api_url = self._get_route()
+        self.headers = self._get_headers()
+        self.scrape_interval = self.get_scrape_interval()
+
+    def _get_route(self):
+        # get route to prometheus HTTP api
+        LOGGER.info("Prometheus: Obtaining route")
+        route = Route(
+            namespace=self.namespace, name=self.resource_name, client=self.client
+        ).instance.spec.host
+
+        return f"https://{route}"
+
+    def _get_headers(self):
+        """Uses the Prometheus serviceaccount to get an access token for OAuth"""
+
+        LOGGER.info("Setting Prometheus headers and Obtaining OAuth token")
+
+        secret = self._get_resource_secret()
+
+        token = secret.instance.metadata.annotations["openshift.io/token-secret.value"]
+
+        return {"Authorization": f"Bearer {token}"}
+
+    def _get_service_account(self):
+        """get service account  for the given namespace and resource"""
+
+        resource_sa = ServiceAccount(
+            namespace=self.namespace, name=self.resource_name, client=self.client
+        )
+        return resource_sa
+
+    def _get_resource_secret(self):
+        """secret for the service account extracted"""
+        resource_sa = self._get_service_account()
+        resource_secret = Secret(
+            namespace=self.namespace,
+            name=resource_sa.instance.imagePullSecrets[0].name,
+            client=self.client,
+        )
+        return resource_secret
+
+    def _get_response(self, query):
+        response = requests.get(
+            f"{self.api_url}{query}", headers=self.headers, verify=self.verify_ssl
+        )
+
+        try:
+            return json.loads(response.content)
+        except JSONDecodeError as json_exception:
+            LOGGER.error(
+                "Exception converting query response to JSON: "
+                f"exc={json_exception} response_status_code={response.status_code} response={response.content}"
+            )
+            raise
+
+    def query(self, query):
+        """
+        get the prometheus query result
+
+        Args:
+            query (str): promthetheus query string
+
+        Returns:
+            dict: query result
+        """
+        return self._get_response(query=f"{self.api_v1}/query?query={query}")
+
+    def get_all_alerts_by_alert_name(self, alert_name):
+        """
+        Get alert by alert name if it's an active alert
+
+        Args:
+             alert (str): alert name
+
+        Examples:
+             result = prometheus.get_alert(alert='WatchDog')
+
+        Returns:
+             list: list containing alert metrics
+        """
+        response = self.alerts()
+        alert_list = []
+        for alert in response["data"]["alerts"]:
+            if alert and alert["labels"]["alertname"] == alert_name:
```

Comment:
`if alert` is not needed, once you inside the loop you have alert.

Reply:
done